### PR TITLE
Remove: mirror middleware and parity comparison with Bose cloud

### LINKED ISSUE #245: [Feature]: remove mirroring
### Library Version

v0.72.0

### Problem Description

With the Bose Soundtouch Cloud being shut down, mirroring isn't of value anymore. The feature can be dropped, the codebase can be cleaned.

### Proposed Solution

Remove that feature.

### Alternatives Considered

_No response_

### Component

Performance Optimization

### Device Compatibility

_No response_

### Priority

Medium - Would improve my workflow

### API Type (if applicable)

None

### Use Case / User Story

.

### Desired API Example

```go

```

### Technical Details

_No response_

### Implementation

- [ ] I can help implement this feature
- [ ] I can provide testing/feedback
- [ ] I can help with documentation
- [ ] I need someone else to implement this

### Research & References

_No response_

### Checklist

- [x] I have searched for existing issues and feature requests
- [x] I have checked the [API Cookbook](docs/API-COOKBOOK.md) for existing functionality
- [x] This feature is related to Bose SoundTouch functionality
- [x] I have considered backward compatibility

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mennemann**
## Description

Removals:

**Mirror middleware**: forwarded requests to the real Bose cloud alongside the local AfterTouch service, then compared responses as "parity mismatches." Became dead code when Bose servers went offline.

**Proxy forwarding to Bose**: previously, unhandled requests and OAuth token failures were silently forwarded to Bose servers. With Bose servers offline, every such request times out. Now: unknown routes return 404, unavailable services return 503, and token refresh failures return 502.

## Type of Change

Please check the type of change your PR introduces:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring (no functional changes)
- [ ] Test improvements
- [ ] Build/CI improvements

## Related Issues

- Closes #245 

### API Changes
- [ ] Added new endpoints
- [x] Modified existing endpoints
- [ ] Added new CLI commands
- [x] Modified existing CLI commands
- [ ] Added new configuration options

### Implementation Details
- Describe the main changes
- List any new dependencies
- Mention any architectural changes

### Automated Tests
- [x] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] All existing tests pass
- [x] Test coverage maintained or improved

### Manual Testing
- [ ] Tested with real SoundTouch device(s)
- [ ] Tested CLI changes manually
- [ ] Tested in different network environments

*not yet*

**Device(s) tested with:**
- Device model: [e.g. SoundTouch 10]
- Device IP: [e.g. 192.168.1.100]
- Test results: [brief description]

*not yet*

### Test Commands
```bash
# Commands used to test this change
make test
go test ./pkg/client -v -run TestNewFeature
soundtouch-cli --host 192.168.1.100 new-command
```

## Documentation

- [ ] Updated relevant documentation
- [ ] Added code comments for complex logic
- [ ] Updated CLI help text
- [ ] Added usage examples
- [ ] Updated API documentation

**Documentation files updated:**
- [ ] README.md
- [ ] docs/API-Endpoints-Overview.md
- [ ] docs/CLI-REFERENCE.md
- [ ] Code documentation (godoc)

## Backward Compatibility

- [ ] This change is backward compatible
- [x] This change includes breaking changes (requires major version bump)
- [ ] This change requires configuration migration

**Breaking changes (if any):**
Now unused fields in `settings.json` are silently ignored, but CLI flags `--mirror-enabled`, `--mirror-endpoints`, `--preferred-source` are gone 

## Security Considerations

- [x] No security implications
- [ ] Security review required
- [ ] Added input validation
- [ ] Updated authentication/authorization

## Performance Impact

- [x] No performance impact
- [ ] Performance improvement
- [ ] Potential performance regression (justify why)

**Performance notes:**
- Measured impact: [benchmarks, timing, memory usage]
- Optimization opportunities: [if any]

## Code Quality

- [x] Code follows project style guidelines
- [x] No linting errors
- [x] No security warnings
- [ ] Memory leaks checked (if applicable)

### Pre-submission Checklist

- [x] `make check` passes (format, lint, vet)
- [x] `make test` passes
- [x] No TODO comments left in production code
- [x] Error handling is comprehensive
- [x] Logging is appropriate (not too verbose, not too quiet)

## Deployment Notes

Any special considerations for deployment:
- Configuration changes required: Users with startup scripts or using `--mirror-enabled`, `--mirror-endpoints`, `--skip-mirror-endpoints`, or `--preferred-source` must remove those flags.
- Database migrations needed: No
- Service restart required: Yes
- Rollback procedures

## Screenshots (if applicable)

If this PR includes UI changes or CLI output changes, include screenshots or terminal output examples.

Mirroring & Parity Tab gone:
image

Mirroring settings gone:
image

---

**Reviewer Guidelines:**
- Check that all tests pass
- Verify documentation is updated
- Test manually if device access available  
- Consider backward compatibility
- Evaluate error handling and edge cases

### CODE REVIEW

**Comment by gesellix** [comment]
> Should the feature that forwards unknown request to Bose servers be removed too?

Yes. Please keep those changes in more or less small commits (build should always work for each commit, lint/tests may be fixed in a final commit on top), so that we have a chance to cherry-pick or at least review each commit separately. In other words: feel free to commit on top as much as you need.

**Comment by gesellix** [comment]
And, because this might become fiddly: when you think the pr is in good shape, make a git rebase, then we can have a clean ci and review without keeping parallel commits in mind. Thanks!

**Comment by mennemann** [comment]
> In other words: feel free to commit on top as much as you need.

So I should keep the current commit as is and not break it down into smaller ones?

**Comment by gesellix** [comment]
> > In other words: feel free to commit on top as much as you need.
> 
> So I should keep the current commit as is and not break it down into smaller ones?

Yes, I'm fine with that.

**Comment by gesellix** [comment]
Sorry for the interrupt: I merged a bigger patch, now we're running into conflicts in main.go. I think it's feasible, please leave a comment if you need any help!

**Comment by mennemann** [comment]
Merge was straightforward.

One more thing: What do we do with `soundtouch-backup`? If we remove the cloud stuff, we are left over with local. Is that fine, or is the tool still needed at all even? (Considering data sync in AfterTouch is a thing)

**Comment by gesellix** [comment]
> What do we do with `soundtouch-backup`?

That's unrelated for now and should also not be affected by the middleware cleanup, right?

**Comment by mennemann** [comment]
Yea, I was just thinking if we should remove the cloud backup as well in this PR, but might make sense to do that separately.

**Comment by mennemann** [comment]
I am pretty sure this is ready for review and testing. I still need to update some documentation files, but I was going to wait with that, until the actual changes are finalized.

**Comment by gesellix** [comment]
> I am pretty sure this is ready for review and testing. I still need to update some documentation files, but I was going to wait with that, until the actual changes are finalized.

Alright, then I'm going to plan for a review tonight 👍

**Review by gesellix — COMMENTED** [review]
This looks really good. I'll make another pass later and let Claude also have a look.
If it's ok for you I would already merge it asap (provided some notes I made for myself won't block anything, and given Claude to not find any breaking change).
The docs update could follow in another pull request?

File: pkg/service/handlers/web/js/script.js [hunk 1]
```diff
@@ -1525,122 +1497,6 @@ async function fetchDeviceEvents(deviceId) {
...
-    if (isLarge && !forceRichDiff) {
-        warningEl.style.display = "block";
-        const forceBtn = document.getElementById("force-rich-diff-btn");
-        forceBtn.onclick = () => viewParityMismatch(m, true);
-
-        document.getElementById("diff-local-body").innerText = localBody;
-        document.getElementById("diff-upstream-body").innerText = upstreamBody;
-    } else {
-        warningEl.style.display = "none";
-        if (typeof Diff !== 'undefined') {
```
Comment (gesellix) [inline pkg/service/handlers/web/js/script.js:1609]: diff.min.js is now obsolete.

File: pkg/service/handlers/handlers_catchall.go
Comment (gesellix) [inline pkg/service/handlers/handlers_catchall.go:1]: Could be renamed to handlers_catchall.go

**Review by gesellix — APPROVED** [review]
Looks good, some details can be addressed in a follow-up.

File: pkg/service/handlers/web/js/script.js (hunk 1 above)
Comment (gesellix) [reply pkg/service/handlers/web/js/script.js:1609]: -> will do.

File: pkg/service/handlers/handlers_catchall.go
Comment (gesellix) [reply pkg/service/handlers/handlers_catchall.go:1]: -> will do.

**Comment by gesellix** [comment]
A fix for the failing Docs Check is planned with https://github.com/gesellix/Bose-SoundTouch/pull/320

**Comment by gesellix** [comment]
Merging, thanks a lot, this cleans the codebase quite a bit!

**Comment by mennemann** [comment]
> The docs update could follow in another pull request?

Should I take care of that?

**Comment by gesellix** [comment]
> > The docs update could follow in another pull request?
> 
> Should I take care of that?

Yes, that would be nice!

I think only these are relevant, maybe you found more places:

- `docs/concepts/technical-specification.md` -> `MirrorEndpoints []string`, `"mirror_endpoints": [...]` JSON
  example.
- `docs/PARITY-IMPROVEMENTS.md` -> references the deleted `TestParityMismatchReproduction_V2/V3` tests.
- `docs/guides/CAPTURE-MIGRATION-TRAFFIC.md` -> documents `GET /setup/parity-mismatches` (the endpoint that's gone).
